src/libcrun/cgroup.c

```diff
@@ -348,7 +348,21 @@ enter_cgroup (int cgroup_mode, pid_t pid, const char *path, int ensure_missing,
       xasprintf (&cgroup_path_procs, "/sys/fs/cgroup/%s/cgroup.procs", path);
       ret = write_file (cgroup_path_procs, pid_str, strlen (pid_str), err);
       if (UNLIKELY (ret < 0))
-        return ret;
+        {
+          if (!ensure_missing && (*err)->status == EBUSY)
+            {
+              /* There are subdirectories so it is not possible to join the initial
```

Comment:
Will this work on cgroups V1?

Reply:
This code runs only on cgroups v2.  There is not such issue on cgroups v1 as a process can be added at any point in the tree, not only to nodes without sub-directories.